js/searchbar/customBangs.js

```diff
@@ -12,8 +12,126 @@ var places = require('places/places.js')
+var searchbarPlugins = require('searchbar/searchbarPlugins.js')
+var tabEditor = require('navbar/tabEditor.js')
+var formatRelativeDate = require('util/relativeDate.js')
+
+function moveToTask(text) {
+  /* disabled in focus mode */
+  if (focusMode.enabled()) {
+    focusMode.warn()
+    return
+  }
+
+  // remove the tab from the current task
+
+  var currentTab = tabs.get(tabs.getSelected())
+  tabs.destroy(currentTab.id)
+
+  // make sure the task has at least one tab in it
+  if (tabs.count() === 0) {
+    tabs.add()
+  }
+
+  var newTask = getTaskByNameOrNumber(text.toLowerCase())
+
+  if (newTask) {
+    newTask.tabs.add(currentTab, { atEnd: true })
+  } else {
+    // create a new task with the given name
+    newTask = tasks.get(tasks.add(undefined, tasks.getIndex(tasks.getSelected().id) + 1))
+    newTask.name = text
+
+    newTask.tabs.add(currentTab)
+  }
+
+  browserUI.switchToTask(newTask.id)
+  browserUI.switchToTab(currentTab.id)
+
+
+  taskOverlay.show()
+
+  setTimeout(function () {
+    taskOverlay.hide()
+  }, 600)
+}
+
+function switchToTask(text) {
+  /* disabled in focus mode */
+  if (focusMode.enabled()) {
+    focusMode.warn()
+    return
+  }
+
+  text = text.toLowerCase()
+
+  // no task was specified, show all of the tasks
+  if (!text) {
+    taskOverlay.show()
+    return
+  }
+
+  var task = getTaskByNameOrNumber(text)
+
+  if (task) {
+    browserUI.switchToTask(task.id)
+  }
+}
+
+// returns a task with the same name or index ("1" returns the first task, etc.)
+// In future PR move this to task.js
+function getTaskByNameOrNumber(text) {
+  const textAsNumber = parseInt(text)
+
+  return tasks.find((task, index) => (task.name && task.name.toLowerCase() === text) || index + 1 === textAsNumber
+  )
+}
+
+// return an array of dicts, sorted by last task activity
+// if a search string is present (and not a number) filter the results with a basic fuzzy search
+function searchAndSortTasks(text) {
+  var sortLastActivity = tasks.map(t => Object.assign({}, { task: t }, { lastActivity: tasks.getLastActivity(t.id) }))
+
+  sortLastActivity = sortLastActivity.sort(function (a, b) {
+    return b.lastActivity - a.lastActivity
+  })
+
+  var isSingleNumber = /^\d+$/.test(text)
+
+  if (text !== '' ? !isSingleNumber : !isSingleNumber) { //lXOR
+    // fuzzy search
+    var matches = []
+    var searchText = text.toLowerCase()
+
+    sortLastActivity.forEach(function (t) {
+
+      var task = t.task
+      var taskName = (task.name ? task.name : l('defaultTaskName').replace('%n', tasks.getIndex(task.id) + 1)).toLowerCase()
+      var exactMatch = taskName.indexOf(searchText) !== -1
+      var fuzzyTitleScore = taskName.score(searchText, 0.5)
+
+      if (exactMatch || fuzzyTitleScore > 0.4) {
+        matches.push({
+          task: t,
+          score: fuzzyTitleScore + exactMatch
+        })
+      }
+    })
+
+    matches = matches.sort(function (a, b) {
+      return b.score - a.score
+    })
```

Comment:
I don't think we should reorder matches based on score - it seems confusing to have tasks be ordered by most-recent most of the time, but occasionally not be. Filtering to include things above the threshold and then sorting by most-recent should be enough.